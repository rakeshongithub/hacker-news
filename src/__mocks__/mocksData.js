export const mockNews = [
  {
    created_at: "2017-11-28T19:41:10.000Z",
    title: "macOS High Sierra: Anyone can login as “root” with empty password",
    url: "https://twitter.com/lemiorhan/status/935578694541770752",
    author: "vladikoff",
    points: 3001,
    story_text: null,
    comment_text: null,
    num_comments: 813,
    story_id: null,
    story_title: null,
    story_url: null,
    parent_id: null,
    created_at_i: 1511898070,
    relevancy_score: 7810,
    _tags: ["story", "author_vladikoff", "story_15800676"],
    objectID: "15800676",
    _highlightResult: {
      title: {
        value:
          "macOS High Sierra: Anyone can login as “root” with empty password",
        matchLevel: "none",
        matchedWords: [],
      },
      url: {
        value: "https://twitter.com/lemiorhan/status/935578694541770752",
        matchLevel: "none",
        matchedWords: [],
      },
      author: {
        value: "vladikoff",
        matchLevel: "none",
        matchedWords: [],
      },
    },
  },
];
