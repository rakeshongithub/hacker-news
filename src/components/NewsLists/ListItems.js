import React from "react";
import styles from "./NewsLists.module.scss";
import ArrowDropUpIcon from "@material-ui/icons/ArrowDropUp";
import LoadingComponent from "../UI/Loading";
import { dateDiff } from "./../../utils/utils";

const ListItems = (props) => {
  const { newsLists = [], isLoading, handleHideItem, handleUpvote } = props;
  return (
    <React.Fragment>
      {isLoading ? (
        <LoadingComponent className={styles.IsLoading} />
      ) : (
        newsLists.map((item, index) => {
          return (
            <div
              className={[styles.ListItems, "d-flex"].join(" ")}
              key={item.objectID}
            >
              <div className={[styles.LeftPanel, "d-flex"].join(" ")}>
                <div className={styles.Comments}>{item.num_comments || 0}</div>
                <div className={styles.UpVotesCount}>
                  {item.points || 0}{" "}
                  <span onClick={() => handleUpvote(index)}>
                    <ArrowDropUpIcon />
                  </span>
                </div>
                <div className={styles.NewsTitle}>
                  <a href={item.url} target="_blank" rel="noopener noreferrer">
                    {item.title}
                  </a>
                </div>
              </div>
              <div className={[styles.RightPanel, "d-flex"].join(" ")}>
                {item.url ? (
                  <div className={styles.LinkedDomain}>
                    ({item.url.split("/")[2]})
                  </div>
                ) : null}
                <div className={styles.AuthorName}>
                  by <span>{item.author}</span>
                </div>
                <div className={styles.TimeDiff}>
                  {dateDiff(new Date(), item.created_at)}
                </div>
                <div className={styles.HideAction}>
                  [ <span onClick={() => handleHideItem(index)}>hide</span> ]
                </div>
              </div>
            </div>
          );
        })
      )}
    </React.Fragment>
  );
};

export default ListItems;
