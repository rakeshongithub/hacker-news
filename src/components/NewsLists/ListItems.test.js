import React from "react";
import { shallow, mount } from "enzyme";
import ListItems from "./ListItems";

import { mockNews } from "./../../__mocks__/mocksData";

describe("List Items Component", () => {
  it("should renders list items component without crashing", () => {
    const wrapper = shallow(<ListItems newsLists={mockNews} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it("should return empty array if no data available", () => {
    const wrapper = shallow(<ListItems />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not break without new list items", () => {
    const wrapper = shallow(<ListItems />);
    expect(wrapper.find("li")).toHaveLength(0);
  });

  it("should render when found new list items", () => {
    const wrapper = mount(<ListItems newsLists={[]} />);

    wrapper.setProps({
      newsLists: mockNews,
    });
    expect(wrapper.find("a").prop("href")).toEqual(
      "https://twitter.com/lemiorhan/status/935578694541770752"
    );
  });
});
