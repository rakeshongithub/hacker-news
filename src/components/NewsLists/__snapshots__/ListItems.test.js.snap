// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Items Component should renders list items component without crashing 1`] = `ShallowWrapper {}`;

exports[`List Items Component should return empty array if no data available 1`] = `ShallowWrapper {}`;
